src/transformers/pipelines.py

```diff
@@ -334,12 +328,11 @@ class Pipeline(_ScikitCompat):
     """
 
     default_input_names = None
-    task = None
 
     def __init__(
         self,
-        model: Optional = None,
-        tokenizer: PreTrainedTokenizer = None,
+        model,
```

Comment:
should we type them to:
```
model: Union[PreTrainedModel, TFPreTrainedModel]
tokenizer: PreTrainedTokenizer
```

Reply:
As seen with @mfuntowicz, if we were to do that it would try to import both `PreTrainedModel` and `TFPreTrainedModel`.

If either TF or PT is unavailable, these objects may not be importable and then they crash. We would need to have both torch and tensorflow as requirements to operate the pipelines were we to type them like that.